http/restful-ws/src/main/java/io/cloudevents/http/restful/ws/impl/BinaryRestfulWSMessageReaderImpl.java

```diff
@@ -19,47 +19,53 @@
 
 import io.cloudevents.SpecVersion;
 import io.cloudevents.core.message.impl.BaseGenericBinaryMessageReaderImpl;
+import io.cloudevents.http.HttpMessageReader;
+import sun.reflect.generics.reflectiveObjects.NotImplementedException;
 
 import javax.ws.rs.core.HttpHeaders;
 import javax.ws.rs.core.MultivaluedMap;
-import java.util.Objects;
+import java.util.*;
 import java.util.function.BiConsumer;
 
 import static io.cloudevents.http.restful.ws.impl.CloudEventsHeaders.CE_PREFIX;
 
-public final class BinaryRestfulWSMessageReaderImpl extends BaseGenericBinaryMessageReaderImpl<String, String> {
+public final class BinaryRestfulWSMessageReaderImpl extends HttpMessageReader {
 
-    private final MultivaluedMap<String, String> headers;
+    static private Iterable<Map.Entry<String, String>> multivaluedMapToIterableHeaders(MultivaluedMap<String, String> multivaluedMap) {
```

Comment:
If the type of the `HttpMessageReader` constructor parameter were `Map<String, String>` then you could use something like this here:

```
multivaluedMap.entrySet().stream().collect(toMap(k -> k, v -> v.getFirst()));
```

(You could also use that with the existing type and then call `entrySet()` on it.)